src/main/scala/Column.scala

```diff
@@ -9,14 +9,512 @@ import com.fasterxml.jackson.databind.node.{
+    "http://www.w3.org/2001/XMLSchema#nonNegativeInteger" -> processNonNegativeInteger(),
+    "http://www.w3.org/2001/XMLSchema#positiveInteger" -> processPositiveInteger(),
+    "http://www.w3.org/2001/XMLSchema#unsignedLong" -> processUnsignedLong(),
+    "http://www.w3.org/2001/XMLSchema#unsignedInt" -> processUnsignedInt(),
+    "http://www.w3.org/2001/XMLSchema#unsignedShort" -> processUnsignedShort(),
+    "http://www.w3.org/2001/XMLSchema#unsignedByte" -> processUnsignedByte(),
+    "http://www.w3.org/2001/XMLSchema#nonPositiveInteger" -> processNonPositiveInteger(),
+    "http://www.w3.org/2001/XMLSchema#negativeInteger" -> processNegativeInteger(),
+    "http://www.w3.org/2001/XMLSchema#double" -> processDoubleDatatype(),
+    "http://www.w3.org/2001/XMLSchema#float" -> processFloatDatatype(),
+    // Date Time related datatype
+    "http://www.w3.org/2001/XMLSchema#date" -> processDateDatatype(),
+    "http://www.w3.org/2001/XMLSchema#dateTime" -> processDateTimeDatatype(),
+    "http://www.w3.org/2001/XMLSchema#dateTimeStamp" -> processDateTimeStamp(),
+    "http://www.w3.org/2001/XMLSchema#gDay" -> processGDay(),
+    "http://www.w3.org/2001/XMLSchema#gMonth" -> processGMonth(),
+    "http://www.w3.org/2001/XMLSchema#gMonthDay" -> processGMonthDay(),
+    "http://www.w3.org/2001/XMLSchema#gYear" -> processGYear(),
+    "http://www.w3.org/2001/XMLSchema#gYearMonth" -> processGYearMonth(),
+    "http://www.w3.org/2001/XMLSchema#time" -> processTime()
+  )
+
+  def trimValue(): (String, Option[String]) => Either[String, String] =
+    (v, _) => Right(v.strip())
+
+  def allValueValid(): (String, Option[String]) => Either[String, String] =
+    (v, _) => Right(v)
+
+  def processBooleanDatatype()
+      : (String, Option[String]) => Either[String, Boolean] = { (v, format) =>
+    {
+      var formatValues = Array[String]()
+      format match {
+        case Some(f) => {
+          formatValues = f.split("""\|""")
+          if (formatValues(0) == v) {
+            Right(true)
+          } else if (formatValues(1) == v) {
+            Right(false)
+          } else Left("invalid_boolean")
+        }
+        case None => {
+          if (Array[String]("true", "1").contains(v)) {
+            Right(true)
+          } else if (Array[String]("false", "0").contains(v)) {
+            Right(false)
+          } else Left("invalid_boolean")
+        }
+      }
+    }
+  }
+
+  def processDecimalDatatype()
+      : (String, Option[String]) => Either[String, Double] = {
+    (value, format) =>
+      {
+        val validDecimalDatatypeRegex =
+          "(\\+|-)?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)".r
+        if (!validDecimalDatatypeRegex.pattern.matcher(value).matches()) {
+          Left("invalid_decimal")
+        } else {
+          numericParser(value, format) match {
+            case Left(_)         => Left("invalid_decimal")
+            case Right(newValue) => Right(newValue.doubleValue())
```

Comment:
I think we should be treating thia as a *Java* BigDecimal instead of as a double. I don't think the [XSD specification](https://www.w3.org/TR/xmlschema11-2/#decimal) limits the precision of decimals, so we will need to be able to cope with values that a double cannot represent.